applications/crossbar/src/modules/cb_user_auth.erl

```diff
@@ -314,50 +341,72 @@ load_md5_results(Context, JObj) ->
     lager:debug("found MD5 credentials belong to user ~s", [wh_doc:id(JObj)]),
     cb_context:set_doc(Context, wh_json:get_value(<<"value">>, JObj)).
 
-%%--------------------------------------------------------------------
+
+%% @public
+-spec cleanup_reset_ids() -> 'ok'.
+cleanup_reset_ids() ->
+    CreatedBefore = wh_util:current_tstamp() - 2 * ?SECONDS_IN_DAY,
+    ViewOptions = [{'startkey', 0}
+                   ,{'endkey', CreatedBefore}
+                   %% TODO: find a limit that matches the number of pwd resets per day
+                   ,{'limit', couch_util:max_bulk_insert()}
+                   ,'include_docs'
+                  ],
+    case couch_mgr:get_results(?WH_ACCOUNTS_DB, ?LIST_BY_MTIME, ViewOptions) of
+        {'error', _E} ->
+            lager:debug("failed to lookup expired reset_ids: ~p", [_E]);
+        {'ok', []} ->
+            lager:debug("no expired reset_ids found");
+        {'ok', UserDocs} ->
+            lager:debug("checking ~b user documents", [length(UserDocs)]),
+            couch_mgr:suppress_change_notice(),
+            ensure_reset_id_deleted(UserDocs),
+            couch_mgr:enable_change_notice(),
+            lager:debug("removed yesterday's expired reset_ids")
+    end.
+
+ensure_reset_id_deleted([Doc|Docs]) ->
+    {'ok', _} = couch_mgr:save_doc(wh_doc:account_db(Doc)
+                                  ,wh_json:delete_key(?RESET_ID, Doc)),
```

Comment:
Curious why you chose not to use `couch_mgr:save_docs/2` instead? Bulk saves are more efficient for Couch to process.